include/dali/core/util.h

```diff
@@ -124,6 +124,18 @@ constexpr std::enable_if_t<std::is_integral<T>::value, T> next_pow2(T n) {
   return pow2;
 }
 
+template <typename T>
+DALI_HOST_DEV DALI_FORCEINLINE
+constexpr std::enable_if_t<std::is_integral<T>::value, T> prev_pow2(T n) {
+  T pow2 = 1;
+  T next = 2;
+  while (n > next) {
+    pow2 = next;
+    next = pow2 * 2;
+  }
+  return pow2;
```

Comment:
This results in an infinite loop if the value of `n` is in the upper half of the range of T.
```suggestion
  T pow2 = 1;
  while (n - pow2 > pow2) {  // use this subtraction to avoid overflow
    pow2 += pow2;
  }
  return pow2;
}
```